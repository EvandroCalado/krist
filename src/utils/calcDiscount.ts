export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
