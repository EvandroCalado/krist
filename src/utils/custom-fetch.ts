import axios from 'axios';

export const customFetch = axios.create({
  baseURL: import.meta.env.VITE_STRAPI_URL,
});
