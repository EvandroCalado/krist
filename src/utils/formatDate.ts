export const formatDate = (dateString: string): string => {
  const date = new Date(dateString);

  return date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};
