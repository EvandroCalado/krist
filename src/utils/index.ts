export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
