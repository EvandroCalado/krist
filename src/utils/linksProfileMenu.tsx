import { Heart, MapPin, Package, Settings, UserRound } from 'lucide-react';

export const linksProfileMenu = [
  {
    id: 1,
    label: 'informações pessoais',
    path: 'profile',
    icon: <UserRound size={16} />,
  },
  {
    id: 2,
    label: 'meus pedidos',
    path: 'orders',
    icon: <Package size={16} />,
  },
  {
    id: 3,
    label: 'lista de desejos',
    path: 'wishlist',
    icon: <Heart size={16} />,
  },
  {
    id: 4,
    label: 'meus endereços',
    path: 'addresses',
    icon: <MapPin size={16} />,
  },
  {
    id: 5,
    label: 'configurar',
    path: 'settings',
    icon: <Settings size={16} />,
  },
];
