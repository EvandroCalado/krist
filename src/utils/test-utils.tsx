import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
