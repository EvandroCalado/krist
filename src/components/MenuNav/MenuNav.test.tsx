import { fireEvent, screen } from '@testing-library/react';

import { customRender } from 'utils/test-utils';

import { MenuNav } from '.';

describe('MenuNav', () => {
  it('should render default', () => {
    customRender(<MenuNav />);

    const nav = screen.getByRole('navigation');
    const links = screen.getAllByRole('link');

    expect(nav).toBeInTheDocument();
    expect(links).toHaveLength(3);
    expect(links[0]).toHaveTextContent('home');
    expect(links[0]).toHaveClass('active');
    expect(links[1]).toHaveTextContent('loja');
    expect(links[2]).toHaveTextContent('contato');

    fireEvent.click(links[1]);

    expect(links[0]).not.toHaveClass('active');
    expect(links[1]).toHaveClass('active');

    fireEvent.click(links[2]);

    expect(links[1]).not.toHaveClass('active');
    expect(links[2]).toHaveClass('active');
  });

  it('should render a snapshot', () => {
    const { container } = customRender(<MenuNav />);

    expect(container).toMatchSnapshot();
  });
});
