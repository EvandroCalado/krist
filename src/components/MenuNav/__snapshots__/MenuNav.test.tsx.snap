// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuNav > should render a snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  gap: 1.6rem;
}

.c1 {
  position: relative;
  color: #131118;
  text-transform: capitalize;
  border-bottom: 2px solid transparent;
  transition: all 150ms ease-in-out;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: -2px;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #131118;
  transition: all 150ms ease-in-out;
}

.c1:hover::after {
  width: 100%;
  left: 0;
}

.c1.active {
  border-bottom: 2px solid #131118;
}

@media (max-width: 768px) {
  .c0 {
    flex-direction: column;
  }
}

<div>
  <nav
    class="c0 menu-nav"
  >
    <a
      aria-current="page"
      class="c1 active"
      href="/"
    >
      home
    </a>
    <a
      class="c1"
      href="/shop"
    >
      loja
    </a>
    <a
      class="c1"
      href="/contact"
    >
      contato
    </a>
  </nav>
</div>
`;
